src/Chainweb/BlockHeader.hs

```diff
@@ -694,6 +695,43 @@ data NewMinedBlock = NewMinedBlock
     deriving (Eq, Show, Generic)
     deriving anyclass (ToJSON, NFData)
 
+data AmberdataBlock = AmberdataBlock
```

Comment:
I think this data structure should go into a different module. What about maybe `Chainweb.Logging.Amberdata`?

`Chainweb.BlockHeader` is a core consensus data structure, and we should only include what is needed for consensus.

(The same holds for the `NewMinedBlock` type.)

Reply:
Agreed.